### PR TITLE
Update Connector to support _ in the broker uri

### PR DESCRIPTION
**Pull request by campru**
Support underscore in the failover host name and parameters in the url. This issue provokes that if you are using docker and the systems are generated automatically it doesn't work because of the _

### CODE REVIEW

**Comment by staabm** [comment]
could you add a unit test with a example of such a hostname to prevent regressions in the future?

**Comment by campru** [comment]
Yes, no problem. I have already added a UT where the host contains underscore

File: src/Stomp/Network/Connection.php [hunk 1]
```diff
@@ -125,7 +125,7 @@ public function __construct($brokerUri, $connectionTimeout = 1, array $context =
         $this->parser = new Parser();
         $this->connectTimeout = $connectionTimeout;
         $this->context = $context;
-        $pattern = "|^(([a-zA-Z0-9]+)://)+\(*([a-zA-Z0-9\.:/i,-]+)\)*\??([a-zA-Z0-9=&]*)$|i";
+        $pattern = "|^(([a-zA-Z0-9]+)://)+\(*([a-zA-Z0-9\.:/i,-_]+)\)*\??([a-zA-Z0-9=&_]*)$|i";
```
Comment (staabm) [inline src/Stomp/Network/Connection.php]: you also added the `_` as allowed character within connection properties... was it intended?

Comment (staabm) [inline src/Stomp/Network/Connection.php]: @jmglsn why do we allow the protocol `(([a-zA-Z0-9]+)://)+` once or more (trailling `+`)?

Comment (staabm) [inline src/Stomp/Network/Connection.php]: in other words, your unit test should also pass when only adding the first underscore

Comment (campru) [inline src/Stomp/Network/Connection.php]: Sorry, I was not intented. I did another commit to revert this change

**Comment by staabm** [comment]
the change looks good to me, thx.

**Comment by campru** [comment]
Thank you. I would like to know one this change is ready to be merged into master, how do you manage to generate a new tag with these changes? Thank you to take a look so fast :)

**Comment by staabm** [comment]
Either @jmglsn has a few mintues to give his opinion within the next few days or I will take care of it at the end of this week.

In the meantime you could use your own fork[1] in case you rely on this change in your project

[1] http://stackoverflow.com/questions/13498519/how-to-require-a-fork-with-composer

**Comment by campru** [comment]
Thank you!

**Comment by jmglsn** [comment]
@campru Thank you for this PR!
@staabm I'll back port this for the version 3.0 branch as well. The allowed `+` can be dropped, but this is ok for the moment.

**Comment by jmglsn** [comment]
It's included in [4.2.1](https://github.com/stomp-php/stomp-php/releases/tag/4.2.1).

**Comment by campru** [comment]
Thank you very much!
